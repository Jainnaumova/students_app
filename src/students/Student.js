import React from 'react';

export default props => {
  const { student } = props;
  return (
    <div className='student-table'>
      <div>{student.studentId}</div>
      <div className='student-name'>
        {`${student.firstName} ${student.lastName}`}
        <div className='details-container'>
          <button className='button' onClick={() => props.displayStudentData(student)}>Details...</button>
        </div>
      </div>
      <div>{student.grade}</div>
      <div className='school-name'>{student.schoolName}</div>
      <div>{student.attendancePercentage}</div>
    </div>
  );
}
