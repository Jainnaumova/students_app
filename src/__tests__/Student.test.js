import React from 'react';
import ReactDOM from 'react-dom';
import Adapter from 'enzyme-adapter-react-15';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import Student from '../students/Student';

Enzyme.configure({ adapter: new Adapter() });

const props = {
  student: {
    studentId: 123,
    firstName: 'John',
    lastName: 'Doe',
    grade: 'CL-101',
    schoolName: 'PS52',
    attendancePercentage: 0.94
  },
  displayStudentData: jest.fn()
}

describe('Students', () => {

  it('should match the shapshot', () => {
    const tree = renderer.create(
      <Student { ...props }/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call displayStudentData from parent component with correct argument', () => {
    const wrapper = shallow(<Student {...props} />);
    const button = wrapper.find('button');
    button.simulate('click');
    expect(wrapper.instance().props.displayStudentData).toHaveBeenCalledWith(props.student);
    expect(wrapper.instance().props.displayStudentData).toHaveBeenCalledTimes(1);
  });

});
