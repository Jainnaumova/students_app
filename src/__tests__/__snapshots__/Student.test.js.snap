// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Students should match the shapshot 1`] = `
<div
  className="student-table"
>
  <div>
    123
  </div>
  <div
    className="student-name"
  >
    John Doe
    <div
      className="details-container"
    >
      <button
        className="details-button"
        onClick={[Function]}
      >
        Details...
      </button>
    </div>
  </div>
  <div>
    CL-101
  </div>
  <div
    className="school-name"
  >
    PS52
  </div>
  <div>
    0.94
  </div>
</div>
`;
